packages/notebook/src/actions.tsx

```diff
@@ -1966,6 +2000,19 @@ namespace Private {
     activeCell: Cell | null;
   }
 
+  export function isNotebookRendered(notebook: Notebook): boolean {
+    if (notebook.cellsToRender.size !== 0) {
+      showDialog({
+        body: `Notebook is still rendering and has for now ${notebook.cellsToRender.size} remaining cells to render.
+
+Please wait for the complete rendering before invoking that action.`,
+        buttons: [Dialog.okButton({ label: 'Ok' })]
+      }).then(() => undefined);
```

Comment:
It makes more sense to add the `catch` clause than an empty `then` clause.

```suggestion
      }).catch((reason) => { console.error('An error occurred when displaying notebook rendering warning', reason); });
```